app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,21 @@
+class CommentsController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+    new_comment = current_user.comments.new(
+      text: comment_params,
+      author_id: current_user.id,
+      post_id: @post.id
+    )
+    if new_comment.save
+      redirect_to "/users/#{@post.author_id}/posts/#{@post.id}", notice: 'Success!'
+    else
+      render :new, alert: 'Error occured!'
```

Comment:
- [ ] Also please add a new definition for the new comments.

Reply:
Hi @dasileker, thanks for the review. However, I don't understand what you mean by new definition for the new comments. Could you be a bit more specific?

Reply:
@dasileker, the app allows for a user to create a comment if that's what you're talking about

Reply:
hello @d2ndjim I mean by adding a new def for the comment `def new`

Reply:
@dasileker, i didn't add the def new for comments because i'm not adding a new view for comments. It's being handled by the user_posts page. This allows a user to be able to create a new comment albeit being a comment for that particular post

Reply:
@dasileker . Please refer to the post show views to grasp what I'm talking about.

Reply:
OK, I get it but you need to fix the views,  right now the data is not stored to be rendered.